src/language-js/printer-estree.js

```diff
@@ -647,7 +647,16 @@ function printPathNoParens(path, options, print, args) {
       ]);
     case "FunctionDeclaration":
     case "FunctionExpression":
-      parts.push(printFunctionDeclaration(path, print, options));
+      parts.push(
+        printFunctionDeclaration(
+          path,
+          print,
+          options,
+          args &&
+            args.expandLastArg &&
+            path.getParentNode().arguments.length > 1
```

Comment:
```suggestion
            getCallArguments(path.getParentNode()).length > 1
```

Reply:
We don't have such a function.

Reply:
But I see what you mean...

**Prettier pr-9662**
[Playground link](https://deploy-preview-9662--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBDAvASwLYAcIBOMAFAGYCuUYMm0xAlMAL70gA0IEuN0AzsqFQECEAO4AFIQn4pUAG1GoAnvw4AjAqjABrODADKuLZigBzZDALk4HONjVwAJo6cAZVGfKpTcAGKFsVBgaM2Q0chgIdhAACxhsOQB1GMx4XiMwOH1pVMwAN1SlMLBeVRATXjgicU1TQORSeUqOACteAA8AIU0dPX1UbDhXEzgGppsQNvb9E1M5OABFcgh4MblmkCMCSoIwmCVcOF4wAkxuaNxT2ETMRxgY5AAOAAYOS4hKxM1cMMujqryow4AEdlvAalwZGheABaKBwJxOaIEOCgzAomreepIRrrCaVbCYCxWfGzeZLFajHHjDgwVBqG53B5IABMtM0mDkswAwhBsNiQEcAKzRciVAAq9JkuI2eWsAEkoC5YPoTmcYABBJX6fbzNaVJhMIA)
```sh
--parser typescript
```

**Input:**
```tsx
a=import(function(){})
```

**Output:**
```tsx
TypeError: Cannot read property 'length' of undefined
    at https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:230593
    at Object.print (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:270143)
    at Ri (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:115017)
    at https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:114478
    at Object.printComments (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:110760)
    at e (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:114458)
    at https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:203119
    at Ti.call (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:112552)
    at f (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:203110)
    at Ti.call (https://deploy-preview-9662--prettier.netlify.app/lib/standalone.js:15:112552)
```

Reply:
Yes we do... https://github.com/prettier/prettier/pull/9608